ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -493,4 +508,37 @@ private void displayWelcomeMessage() {
 
     writer.flush();
   }
+
+  static void maybeCreateProcessingLogStream(
+      final ProcessingLogConfig config,
+      final KsqlConfig ksqlConfig,
+      final KsqlEngine ksqlEngine,
+      final CommandQueue commandQueue) {
+    if (!config.getString(ProcessingLogConfig.STREAM_AUTO_CREATE).equals(
+        ProcessingLogConfig.AUTO_CREATE_ON)) {
+      return;
+    }
+    final String name = config.getString(ProcessingLogConfig.STREAM_NAME);
+    final String topicName = config.getString(ProcessingLogConfig.TOPIC_NAME);
+    final String statementText = ProcessingLogServerUtils.processingLogStreamCreateStatement(
+        name,
+        topicName);
+    if (!commandQueue.empty()) {
```

Comment:
why do we only create the log stream if the command queue is empty? EDIT: the description of the commit addresses this, maybe an inline comment would help